src/app/product-list/product/product.component.ts

```diff
@@ -0,0 +1,44 @@
+
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
+import { ApidataService } from 'src/app/Services/apidata.service';
+import { CartService } from 'src/app/Services/cart.service';
+import { DataService } from 'src/app/Services/data.service';
+
+
+@Component({
+  selector: 'app-product',
+  templateUrl: './product.component.html',
+  styleUrls: ['./product.component.scss']
+})
+export class ProductComponent implements OnInit, OnChanges {
+
+  @Input() products;
+  @Input() filteredProducts;
+  displayProducts;
```

Comment:
If we're taking an array of products shouldn't this be a productList component instead?